### PR TITLE
Pulling Expr to Subfield logic outside the Tests

### PR DESCRIPTION
**Pull request by beroyfb**
Summary: The logic used for converting expression by the PlanBuilder can be used by the engnies for parts of the filter pushdowmn implementation. So it makes sense to pulling to a more public location.

Differential Revision: D37087713

### CODE REVIEW

File: velox/expression/ExprToSubfieldFilter.h [hunk 1]
```diff
@@ -15,9 +15,16 @@
  */
 #pragma once
 
-#include "velox/type/Filter.h"
+#include <velox/core/Expressions.h>
+#include <velox/core/ITypedExpr.h>
+#include <velox/core/PlanFragment.h>
```
Comment (mbasmanova) [inline velox/expression/ExprToSubfieldFilter.h:20]: This include is probably not needed. Same for PlanNode.h.

File: velox/expression/ExprToSubfieldFilter.h [hunk 2]
```diff
@@ -15,9 +15,16 @@
...
+#include <velox/core/PlanNode.h>
+#include <velox/type/Filter.h>
+#include <velox/type/Subfield.h>
 
-namespace facebook::velox::common::test {
+using namespace facebook::velox;
```
Comment (mbasmanova) [inline velox/expression/ExprToSubfieldFilter.h:25]: Do not use "using" in header files.

File: velox/expression/ExprToSubfieldFilter.h [hunk 3]
```diff
@@ -15,9 +15,16 @@
...
+
+namespace facebook::velox::exec {
```
Comment (mbasmanova) [inline velox/expression/ExprToSubfieldFilter.h:27]: Looks like this should be velox::core namespace, same one as TypedExprPtr. It then makes sense to move this file to 'core' directory. Sorry,  I mistakenly thought that 'expression' directory would be the right one.

**Review by mbasmanova — COMMENTED** [review]
@beroyfb It is nice to reuse this logic in Spark's integration. Thank you for refactoring. It would be helpful to eventually add tests for this functionality. Some comments below.

File: velox/expression/ExprToSubfieldFilter.h (hunk 3 above)
Comment (beroyfb) [reply velox/expression/ExprToSubfieldFilter.h:25]: It cannot live in velox::core. The issue is that it uses Expr and expression uses core. So leads to circular dependency (I tried). So to me seems exec is the right place.

Comment (mbasmanova) [reply velox/expression/ExprToSubfieldFilter.h:23]: Got it. Thank you for explaining.

Comment (mbasmanova) [reply velox/expression/ExprToSubfieldFilter.h:23]: Curious where does the usage of "Expr" is coming from.

File: velox/expression/ExprToSubfieldFilter.h [hunk 4]
```diff
@@ -15,42 +15,45 @@
...
+#include <velox/core/Expressions.h>
+#include <velox/core/ITypedExpr.h>
+#include <velox/type/Filter.h>
+#include <velox/type/Subfield.h>
 
-namespace facebook::velox::common::test {
+namespace facebook::velox::exec {
 
-inline std::unique_ptr<common::BigintRange> lessThan(
+inline std::unique_ptr<velox::common::BigintRange> lessThan(
```
Comment (mbasmanova) [inline velox/expression/ExprToSubfieldFilter.h:25]: Why do we need "velox::" prefix here?

**Review by mbasmanova — APPROVED** [review]
@beroyfb Looks good to me.

File: velox/expression/ExprToSubfieldFilter.h (hunk 4 above)
Comment (beroyfb) [reply velox/expression/ExprToSubfieldFilter.h:25]: Multiple of filter builder functions take Expr as input for example:
```
std::unique_ptr<common::Filter> makeLessThanFilter(
    const core::TypedExprPtr& upperExpr)
```

Comment (mbasmanova) [reply velox/expression/ExprToSubfieldFilter.h]: > Multiple of filter builder functions take Expr as input for example:

They take core::xxxExpr, which are defined in "core", not in "expression".

**Pull request closed without merging**